src/sagemaker/local/image.py

```diff
@@ -461,6 +459,16 @@ def _generate_compose_file(self, command, additional_volumes=None, additional_en
         }
 
         docker_compose_path = os.path.join(self.container_root, DOCKER_COMPOSE_FILENAME)
+
+        try:
+            import yaml
+        except ImportError as e:
+            logging.warning(
+                "yaml failed to import. Local mode features will be impaired or broken."
+            )
+            # Any subsequent attempt to use yaml will raise the ImportError
+            yaml = sagemaker.utils.DeferredError(e)
+
```

Comment:
I wonder if we should refactor this pattern into its own utility function at this point